### ISSUE TITLE
Create Sign-up/sign-in Flow

### ISSUE THREAD

**Issue by GeorgiaHCA** [alan-turing-institute/AutSPACEs#17]
### Summary

Create a flow for user to sign up to platform.

A wireframe MVP of the share experience page: 

[simplified sign up flow](https://www.figma.com/file/HxqTSdeyAUNMualnBV0PET/Autspace-(WIP)?node-id=168%3A1)

### Deadline 

16th Feb 2021

### What needs to be done?
- [ ] After a user successfully signs **up**, direct them to Frame 1
- [ ] Implement Frame 1
    - [ ] Add question "Are you Autistic?"
    - [ ] Add radio buttons with options: Yes, No, Not sure, Prefer not to say
    - [ ] Add `Continue` button that leads to Frame 4 
- [ ]  Implement Frame 4
    - [ ] Add question "If you have another connection to autism, please write it here. For example, sister, researcher, or friend. You can also leave this space blank."
    - [ ] Add radio buttons with options listed in the image below.
    ![CleanShot 2021-06-29 at 07 56 51](https://user-images.githubusercontent.com/1872836/123793336-95db8480-d8af-11eb-8374-b045a3b78850.jpg)

    - [ ] Add `Submit` button that directs user to Confirmation navigation page (if #121 has been implemented) or to the landing page if #121 has not been implemented.  

### Archived 
Leaving this tasks here for posterity. 

- Allow user to create username and password to sign-in to platform
- Must use Open Humans sign in process: see diagram in the [user-flow doc](/user-flow.md)
- Must seek to collect user's autism status (but not mandated requirement) with option to self-describe
- Must allow users to sign in as moderators (links to #100 )

### Who can help

* Anyone with web development skills to build
* Anyone from the community to review and test*
* the designs are already founded on substantial community feedback and user testing

**Comment by GeorgiaHCA** [comment]
Are you able to take this on @sowmya9507 ?

**Comment by GeorgiaHCA** [comment]
<img width="1930" alt="image" src="https://user-images.githubusercontent.com/46889966/119357176-2a463c00-bc9f-11eb-9319-26938bd6c2fc.png">

**Comment by GeorgiaHCA** [comment]
Original user testing feedback

From https://github.com/alan-turing-institute/AutisticaCitizenScience/blob/master/community-recommendations/user-tests/sprint-6-feedback/summary-sprint6-feedback.md

‘How it works’ section

    A C d: “I want to look only at ‘how it works’.”
    A C d: “‘Authorise project’, what does that mean? Is that the consent-thing?”
    A C d: “I would maybe put ‘authorise project’ in brackets and write ‘give consent’, because that describes in plain English what ‘authorise project’ means.”

Being redirected to the Open Humans platform

    d: 3x not confusing, 2x confusing
    A D d: Unclear what Open Humans is and why you need to login there
    A D d: “Having to login to Open Humans makes me a bit nervous. I would like more information before signing up and logging in, because you don’t really know what you are logging in for.”
    A (1) d: “I don’t find it confusing. I like the fact that it allows you to use your username or email. I would recognize these, it is positive.”
    A (1) d: “I believe it [that you need to login via Open Humans] was mentioned beforehand to me.”
    A (2) d: “I struggle with the organisations. I mean there is Turing..I don’t really know what Open Humans is. I have heard that phrase many times, I am sure that I have been told, but I don’t know. It is some kind of platform - but I don’t even know what a platform is, I am not a technical person. So that is a problem. But I would think here, if I had already signed up before, I would just look up the login details in my notes. I wouldn’t know what Open Humans is, but I have heard this terminology before.”

→ Open Humans and why you need to login via this platform should be explained. The fact that the majority of the participants was not confused by the redirection to Open Humans might be explainable by the fact that they had heard about it before in prior contributions to the project.

**Comment by jhlink** [comment]
Very insightful user feedback, @GeorgiaHCA !! 

This ticket is a bit old and needs some updating, which I think was inevitable since the project evolved a bit after the former was created.  

----

So, given the [simplified wireframe](https://www.figma.com/file/HxqTSdeyAUNMualnBV0PET/Autspace-(WIP)?node-id=168%3A1), there should be a screen preceding frame 1, which takes the user to a simple `username`/`password` text box with the OpenHumans logo. There should also be a `Create Open Humans account` option on this screen, too. Currently, the project takes the user to a OH project page, which I'm not sure should be the desired behavior. 
```
Allow user to create username and password to sign-in to platform
Must use Open Humans sign in process: see diagram in the user-flow doc
```

----

Okay. So, this task should be broken down to implement the two questions in Frame 1 and Frame 4. 
```
Must seek to collect user's autism status (but not mandated requirement) with option to self-describe
```

----

Is this talking about Frame 5? The confirmation page that follows the optional autism question status? 
```
 Confirmation and forward navigation
```

----

This should be part of milestone 4. This milestone isn't focused on defining user privileges, which are needed to implement the following. 
```
 Must allow users to sign in as moderators (links to #100 )
```

**Comment by mridubhatnagar** [comment]
I'm working on this issue.

**Comment by jhlink** [comment]
Hi @mridubhatnagar !! 👋 

Out of curiosity, could you share any progress you've made so far? 
I think this task is a bit large for one ticket, and I'd like to simplify it. 

That said, I'd like to understand where you are before I start archiving tasks. :)

**Comment by mridubhatnagar** [comment]
Hey @jhlink 

Sorry for no updates and delayed response.
I have completed the first page of the sign-up flow, I'll push the changes today.

After that, I can start up with the next page.

**Comment by jhlink** [comment]
@mridubhatnagar , just to let you know, any design related changes will be handled in a near future milestone. 

You don't need to change the button appearance for this ticket 😁

**Comment by mridubhatnagar** [comment]
@jhlink thank you for the update.

Based on the revised task description. I have implemented Frame 1 and have raised PR for the same. :)

**Comment by jhlink** [comment]
@mridubhatnagar sorry for the late reply!
Fabulous. Go ahead and implement frame 4 as part of PR #122 !! 
It'd probably be easier to create a new html file for question 2.
